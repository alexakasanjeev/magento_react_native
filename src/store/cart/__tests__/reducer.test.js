import cartReducer, { INITIAL_STATE } from '../reducer';
import {
  MAGENTO,
  ACTION_USER_LOGOUT,
} from '../../../constants';

describe('cart reducer', () => {
  test('should return the initial state', () => {
    // Exercise
    const newState = cartReducer(undefined, {});

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle remove item from cart loading', () => {
    // Setup
    const action = {
      type: MAGENTO.REMOVE_ITEM_FROM_CART_LOADING,
    };

    // Exercise
    const newState = cartReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle remove item from cart failure', () => {
    // Setup
    const action = {
      type: MAGENTO.REMOVE_ITEM_FROM_CART_FAILURE,
      payload: {
        errorMessage: 'Unable to remove item from cart',
      }
    };

    // Exercise
    const newState = cartReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle customer cart loading', () => {
    // Setup
    const action = {
      type: MAGENTO.CUSTOMER_CART_LOADING,
    };

    // Exercise
    const newState = cartReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle customer cart success', () => {
    // Setup
    const action = {
      type: MAGENTO.CUSTOMER_CART_SUCCESS,
      payload: {
        cart: {
          // Dummy cart object(not actual representation)
          mockCartId: 2
        }
      }
    };

    // Exercise
    const newState = cartReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle customer cart failure', () => {
    // Setup
    const action = {
      type: MAGENTO.CUSTOMER_CART_FAILURE,
      payload: {
        errorMessage: 'Unable to access cart',
      }
    };

    // Exercise
    const newState = cartReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle cart item product info success', () => {
    // Setup
    const action = {
      type: MAGENTO.CART_ITEM_PRODUCT_SUCCESS,
      payload: {
        // dummy object(Not actual representation)
        sku: 'WH02-S',
        title: 'Product Title',
        price: 0
      }
    };

    // Exercise
    const newState = cartReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle user logout', () => {
    // Setup
    const action = {
      type: ACTION_USER_LOGOUT,
    };

    // Exercise
    const newState = cartReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });
});
