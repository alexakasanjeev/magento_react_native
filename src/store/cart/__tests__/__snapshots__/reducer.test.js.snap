// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart reducer should handle cart item product info success 1`] = `
Object {
  "cart": Object {},
  "errorMessage": "",
  "products": Object {
    "WH02-S": Object {
      "price": 0,
      "sku": "WH02-S",
      "title": "Product Title",
    },
  },
  "status": "NOT_STARTED",
}
`;

exports[`cart reducer should handle customer cart failure 1`] = `
Object {
  "cart": Object {},
  "errorMessage": "Unable to access cart",
  "products": Object {},
  "status": "ERROR",
}
`;

exports[`cart reducer should handle customer cart loading 1`] = `
Object {
  "cart": Object {},
  "errorMessage": "",
  "products": Object {},
  "status": "LOADING",
}
`;

exports[`cart reducer should handle customer cart success 1`] = `
Object {
  "cart": Object {
    "mockCartId": 2,
  },
  "errorMessage": "",
  "products": Object {},
  "status": "SUCCESS",
}
`;

exports[`cart reducer should handle remove item from cart failure 1`] = `
Object {
  "cart": Object {},
  "errorMessage": "Unable to remove item from cart",
  "products": Object {},
  "status": "ERROR",
}
`;

exports[`cart reducer should handle remove item from cart loading 1`] = `
Object {
  "cart": Object {},
  "errorMessage": "",
  "products": Object {},
  "status": "LOADING",
}
`;

exports[`cart reducer should handle user logout 1`] = `
Object {
  "cart": Object {},
  "errorMessage": "",
  "products": Object {},
  "status": "NOT_STARTED",
}
`;

exports[`cart reducer should return the initial state 1`] = `
Object {
  "cart": Object {},
  "errorMessage": "",
  "products": Object {},
  "status": "NOT_STARTED",
}
`;
